File: app/src/main/java/com/hsilighting/pagify/ui/components/NewDocumentChooser.kt
package com.hsilighting.pagify.ui.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.FolderOpen
import androidx.compose.material.icons.filled.NoteAdd
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp

/**
 * The two things + can mean.
 *
 * Asked rather than assumed, because they are not variations of one action:
 * opening a file you have and making paper you do not are different intentions
 * that happen to start from the same button.
 */
@Composable
fun NewDocumentChooser(
    onBlankPages: () -> Unit,
    onOpenFile: () -> Unit,
    onDismiss: () -> Unit,
) {
    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Add a document") },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                Choice(
                    icon = Icons.Filled.NoteAdd,
                    title = "Blank pages",
                    detail = "Paper to write on: how many, what size and colour",
                    onClick = onBlankPages,
                )
                Choice(
                    icon = Icons.Filled.FolderOpen,
                    title = "Open a file",
                    detail = "A PDF already on this phone or in your storage",
                    onClick = onOpenFile,
                )
            }
        },
        // No confirm button: both answers are in the list, and a dialog whose
        // real choices sit above an OK invites people to press the OK.
        confirmButton = {},
        dismissButton = { TextButton(onClick = onDismiss) { Text("Cancel") } },
    )
}

@Composable
private fun Choice(
    icon: ImageVector,
    title: String,
    detail: String,
    onClick: () -> Unit,
) {
    Surface(
        color = MaterialTheme.colorScheme.surfaceVariant,
        shape = RoundedCornerShape(14.dp),
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick),
    ) {
        Row(
            Modifier.padding(horizontal = 14.dp, vertical = 12.dp),
            horizontalArrangement = Arrangement.spacedBy(14.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Icon(icon, contentDescription = null, tint = MaterialTheme.colorScheme.primary)
            Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                Text(title, style = MaterialTheme.typography.titleSmall)
                Text(
                    detail,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
        }
    }
}

File: ios/Sources/NewDocumentChooser.swift
import SwiftUI

/// The two things `+` can mean.
///
/// Asked rather than assumed, because they are not variations of one action:
/// opening a file you have and making paper you do not are different intentions
/// that happen to start from the same button.
struct NewDocumentChooser: View {
    let onBlankPages: () -> Void
    let onOpenFile: () -> Void
    let onDismiss: () -> Void

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text("Add a document")
                .font(.title3.weight(.semibold))
                .padding(.bottom, 16)

            VStack(spacing: 10) {
                Choice(icon: "doc.badge.plus",
                       title: "Blank pages",
                       detail: "Paper to write on: how many, what size and colour",
                       action: onBlankPages)
                Choice(icon: "folder",
                       title: "Open a file",
                       detail: "A PDF already on this phone or in your storage",
                       action: onOpenFile)
            }

            // No confirm button: both answers are in the list, and a dialog whose
            // real choices sit above an OK invites people to press the OK.
            HStack {
                Spacer()
                Button("Cancel") { onDismiss() }
            }
            .padding(.top, 16)
        }
        .padding(20)
        .background(PagifyColor.background(scheme))
    }
}

private struct Choice: View {
    let icon: String
    let title: String
    let detail: String
    let action: () -> Void

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        Button(action: action) {
            HStack(spacing: 14) {
                Image(systemName: icon)
                    .font(.system(size: 20))
                    .foregroundStyle(PagifyColor.primary(scheme))
                VStack(alignment: .leading, spacing: 2) {
                    Text(title)
                        .font(.subheadline.weight(.semibold))
                        .foregroundStyle(PagifyColor.onSurface(scheme))
                    Text(detail)
                        .font(.caption)
                        .foregroundStyle(.secondary)
                        .multilineTextAlignment(.leading)
                }
                Spacer(minLength: 0)
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 12)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(PagifyColor.surfaceVariant(scheme), in: RoundedRectangle(cornerRadius: 14))
        }
        .buttonStyle(.plain)
    }
}
